### PR TITLE
15: Edit Listing

### LINKED ISSUE #15: Edit Listing
Request Body
- image
- name
- description
- price
- street
- city
- state
- zipcode

### PR DESCRIPTION
**Pull request by LDomadia**
Works similar to "Create Listing"
Differences:
- checks if listing belongs to the user that is passed
- adjusted duplicate address check, just in case address remains unchanged in the edits

### CODE REVIEW

File: server/src/routes/listings.ts [hunk 1]
```diff
@@ -128,10 +128,12 @@ listingRoutes.post('/create', upload.array('imageArray[]'), async (req: Request,
     validation.validateCity(city);
     validation.validateState(state);
     validation.validateZip(zipcode);
+    validation.validLatitude(lat);
+    validation.validLongitude(lon);
     validation.validUID(ownerId);
```
Comment (uityh) [inline server/src/routes/listings.ts]: needs an await in front of it

File: server/src/routes/listings.ts [hunk 2]
```diff
@@ -164,6 +167,45 @@ listingRoutes.get("/:listingId", async (req: Request, res: Response) => {
...
+  try {
+    validation.validString(description);
+    validation.validPrice(price);
+    validation.validString(street);
+    validation.validateCity(city);
+    validation.validateState(state);
+    validation.validateZip(zipcode);
+    validation.validLatitude(lat);
+    validation.validLongitude(lon);
+    validation.validUID(ownerId);
```
Comment (uityh) [inline server/src/routes/listings.ts]: also needs an await

**Review by uityh — APPROVED** [review]
Looks good! Only thing is that the awaits need to be added to the validUID calls since it is an asynchronous function.

File: server/src/routes/listings.ts (hunk 1 above)
Comment (LDomadia) [reply server/src/routes/listings.ts]: fixed!

File: server/src/routes/listings.ts (hunk 2 above)
Comment (LDomadia) [reply server/src/routes/listings.ts]: fixed!

**Comment by cruiznunez** [comment]
please bring branch up to date for review :)

**Comment by LDomadia** [comment]
> please bring branch up to date for review :)

Merge conflicts resolved

**Review by davidtayloryangg — APPROVED** [review]
its working great good stuff

**Pull request merged**